tests/unit/AgreementType.spec.ts

```diff
@@ -0,0 +1,116 @@
+import Vue from 'vue'
+import Vuetify from 'vuetify'
+import { mount, Wrapper, createLocalVue } from '@vue/test-utils'
+import flushPromises from 'flush-promises'
+import VueRouter from 'vue-router'
+import mockRouter from './MockRouter'
+
+import { AgreementType } from '@/components/IncorporationAgreement'
+
+// Store
+import { getVuexStore } from '@/store'
+
+Vue.use(Vuetify)
+
+let vuetify = new Vuetify({})
+const store = getVuexStore()
+
+// Input field selectors to test changes to the DOM elements.
+const sampleTypeSelector: string = '#agreement-type-sample'
+const customTypeSelector: string = '#agreement-type-custom'
+const summaryErrorMessageSelector: string = '.agreement-invalid-message'
+const summaryTextSelector: string = '.summary-desc'
+
+/**
+ * Utility method to get around with the timing issues
+ */
+async function waitForUpdate (wrapper: Wrapper<Vue>) {
+  await wrapper.vm.$nextTick()
+  await flushPromises()
+  await wrapper.vm.$nextTick()
+}
+
+/**
+ * Creates and mounts a component, so that it can be tested.
+ *
+ * @returns a Wrapper<AgreementType> object with the given parameters.
+ */
+function createComponent (showErrorSummary:boolean = false, isSummary:boolean = false): Wrapper<AgreementType> {
+  const localVue = createLocalVue()
+  localVue.use(VueRouter)
+  const router = mockRouter.mock()
+  localVue.use(Vuetify)
+  document.body.setAttribute('data-app', 'true')
+  return mount(AgreementType, {
+    localVue,
+    propsData: {
+      showErrorSummary: showErrorSummary,
+      isSummary: isSummary
+    },
+    router,
+    store,
+    vuetify
+  })
+}
+
+store.state.stateModel.nameRequest.entityType = 'BCOMP'
```

Comment:
Actually, this should be 'BC'.
Please also fix in OrgPerson.spec.ts and ShareStructure.spec.ts.